src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.cs

```diff
@@ -428,7 +428,15 @@ public NamedTypeSymbol Retarget(NamedTypeSymbol type, RetargetOptions options)
             {
                 if (type.IsTupleType)
                 {
-                    return ((TupleTypeSymbol)type).WithUnderlyingType(Retarget(type.TupleUnderlyingType, options));
+                    var newUnderlyingType = Retarget(type.TupleUnderlyingType, options);
+                    if (newUnderlyingType.IsErrorType())
```

Comment:
>newUnderlyingType.IsErrorType() [](start = 24, length = 31)

It should be fine to call WithUnderlyingType with some error types, in fact the original underlying type can be an error type. We should probably use IsTupleOrCompatibleWithTupleOfCardinality check here instead.